### PR TITLE
docs(java): Add SoftAssertions for Java

### PR DESCRIPTION
**Pull request by uchagani**
This PR adds the `SoftAssertions` API for Java.  Related PR: https://github.com/microsoft/playwright-java/pull/1340

Note: https://github.com/microsoft/playwright.dev/pull/1135 needs to be merged in order for the markdown in this PR to be rendered without errors

cc: @yury-s

### CODE REVIEW

File: docs/src/api/class-softassertions.md
Comment (yury-s) [inline docs/src/api/class-softassertions.md:107]:
```suggestion
Runs after all the assertions have been executed for this [SoftAssertions] object.  If any assertions failed, this method throws an AssertionFailedError with the details of all the failed assertions.
```

**Review by yury-s — APPROVED** [review]

**Comment by yury-s** [comment]
Once this lands, let's [roll new driver](https://github.com/microsoft/playwright-java/blob/main/ROLLING.md) with the new API as a separate PR keeping all new methods unimplemented and then apply your implementation on top of it in a separate PR.

**Comment by yury-s** [comment]
I'll commit this right-away to unblock you. Feel free to address any comments in a follow-up PR.

File: docs/src/api/class-softassertions.md
Comment (uchagani) [reply docs/src/api/class-softassertions.md:36]: I have the implementation of these already worked out but i haven't committed it yet.  If it's okay with you, i think we can do everything together.

Comment (yury-s) [reply docs/src/api/class-softassertions.md:36]: Sounds good.

**Pull request merged**
